Tools/YamlCreate.ps1

```diff
@@ -529,14 +543,21 @@ Function Get-InstallerFile {
 
   # Create a new web client for downloading the file
   $_WebClient = [System.Net.WebClient]::new()
-  $_WebClient.Headers.Add('User-Agent', 'Microsoft-Delivery-Optimization/10.1')
+  $_WebClient.Headers.Add('User-Agent', $script:UserAgent)
   # If the system has a default proxy set, use it
   # Powershell Core will automatically use this, so it's only necessary for PS5
   if ($PSVersionTable.PSVersion.Major -lt 6) { $_WebClient.Proxy = [System.Net.WebProxy]::GetDefaultProxy() }
   # Download the file
-  $_WebClient.DownloadFile($URI, $_OutFile)
-  # Dispose of the web client to release the resources it uses
-  $_WebClient.Dispose()
+  try {
+    $_WebClient.DownloadFile($URI, $_OutFile)
+  } catch {
+    # Failed to download with the Delivery-Optimization User Agent, so try again with the WinINet User Agent
+    $_WebClient.Headers.Add('User-Agent', $script:backupUserAgent)
```

Comment:
Do we need to clear the previous `User-Agent` from the Headers list? It appears this just appends it at the end

![image](https://github.com/microsoft/winget-pkgs/assets/88161975/995f36c8-ad67-4e15-80d8-9ce68ac69ab2)

Reply:
> Do we need to clear the previous `User-Agent` from the Headers list? It appears this just appends it at the end
> 
> ![image](https://private-user-images.githubusercontent.com/88161975/317826780-995f36c8-ad67-4e15-80d8-9ce68ac69ab2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2NDg3ODksIm5iZiI6MTcxMTY0ODQ4OSwicGF0aCI6Ii84ODE2MTk3NS8zMTc4MjY3ODAtOTk1ZjM2YzgtYWQ2Ny00ZTE1LTgwZDgtOWNlNjhhYzY5YWIyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI4VDE3NTQ0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk3ZjE1YzhkNjQwNTE5YzBkMTJjMzlkMjk2OTA2NWFjYTEzNmFhMmRmMDNjZjUzZmZiOWJlNGMyOGI2OTcyMzgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.aaAijnj_wVkTqsS20cpQ7D8jkCKcK5NFGHsVdkrj4lU)

When I tested, it seemed that the new request took the latest user header. It probably wouldn't hurt to clear it though

Reply:
Got a chance to test it in Fiddler, and it appears it wasn't using the agent correctly before

![image](https://github.com/microsoft/winget-pkgs/assets/88161975/40fa8413-245d-4f00-b445-01a682a444e2)